compiler/rustc_errors/src/emitter.rs

```diff
@@ -2824,6 +2829,19 @@ impl Style {
     }
 }
 
+/// Whether the original and suggested code are the same.
+pub fn is_no_change(sm: &SourceMap, suggested: &str, sp: Span) -> bool {
```

Comment:
nit: maybe instead of `is_no_change` and then negating again at the callsite (`!is_no_change` is a bit confusing), perhaps call this `is_different` or something like that?